sparkly/utils.py

```diff
@@ -93,6 +181,45 @@ def kafka_get_topics_offsets(host, topic, port=9092):
     return offsets
 
 
+class lru_cache(object):
```

Comment:
Sure.

First example could be a transformation that is used multiple times in a Spark project. More concretely, let's say you have 
```
def transformation_a(...):
    # return some dataframe
```
and let's say `transformation_a` is called twice from different parts of the code. To cache the result you could just decorate it with `@lru_cache` and call it a day instead of handling it explicitly anywhere in code.

Another example would be the following. Think that you have a `settings.yaml` where you define your data sources:
```
DATA_SOURCES:
    table_a: 
        url: 'mysql://blahblah'
        cache: True

    table_b:
        url: 'parquet:s3://blahblah'
        cache: False
```
and we have a reader that loads the dataframes from the `url` sources. Here we might want to cache the MySQL table because it's read in multiple independent transformations but we don't want to keep hitting the hot storage to get it. Now again you could just decorate with `@lru_cache` the main reader code and not have to do anything else in code.

In our private code these examples correspond to https://github.com/Tubular/tbcode/blob/master/svc-creator/loader/src/creator/loader/transformations/accounts/followers_growth.py#L193-L194 and the latter is there: https://github.com/Tubular/tbcode/blob/master/svc-creator/loader/src/creator/loader/shared/io.py#L162-L229